src/libstd/sync/mutex.rs

```diff
@@ -252,6 +253,16 @@ impl<T: Send + 'static> Drop for Mutex<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T: fmt::Debug + Send + 'static> fmt::Debug for Mutex<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.try_lock() {
+            Ok(guard) => write!(f, "Mutex {{ data: {:?} }}", *guard),
+            Err(_) => write!(f, "Mutex {{ <locked> }}")
```

Comment:
This could also test for each variant of `Err` as one case is that the mutex is already locked but another case is that the mutex is poisoned (but the lock was successfully acquired).